src/utils/claim.ts

```
@@ -51,9 +53,19 @@ const createAdjustedClaim = async <
     for (const details of claimDetails) {
         inputSum += await getOutputAmount(asset, details);
     }
+
+    if (typeof inputSum !== "number") {
+        log.error(
+            "inputSum is not valid. Expected number, got",
+            typeof inputSum,
+        );
+        throw new Error("inputSum is not valid");
+    }
```

In what scenario can this happen?

---

@michael1011 when `swapOutput` is undefined (`const swapOutput = detectSwap(tweakedKey, lockupTx);`), this causes `inputSum` to be NaN (which ironically is of type `number`, so this wouldn't work properly indeed).

The reason I added this in first place was because I was having an error difficult to interpret "down the road", which was caused by a NaN fee variable.

I thought having this here would help with developer experience, but actually I think it would make more sense to throw this error even earlier:
```
    const swapOutput = detectSwap(tweakedKey, lockupTx);

    if (swapOutput === undefined) {
        throw new Error("Swap output is undefined");
    }
```
The root cause of my problem was having `swapOutput` undefined due to passing the wrong rescue key to it (was importing the rescueKey from the GlobalContext instead of the one set by the user on the RescueContext).

IG it would be better to throw this error immediately instead of letting it break 3 functions later.